lib/repository/villager_details.dart

```diff
@@ -0,0 +1,14 @@
+import 'package:acnhpedia/api_key.dart';
+import 'package:acnhpedia/networking/api_base_helper.dart';
+import 'package:acnhpedia/models/villager_model.dart';
+
+class VillagerDetailRepository {
+  final String _apiKey = api_key;
+
+  ApiBaseHelper _helper = ApiBaseHelper();
+
+  Future<villager> fetchVillagerData(String villager_name) async {
```

Comment:
use camel case for `villagerName`
the method expects `Future<Villager>` but what will happen if the response throws an error?